implementations/micrometer-registry-new-relic/src/main/java/io/micrometer/newrelic/ClientProviderType.java

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright 2020 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.newrelic;
+
+public enum ClientProviderType {
+    insights_api, insights_agent
```

Comment:
Let's make these uppercase to match the naming convention for enums and let's leniently parse case insensitive when converting from String to enum

Reply:
Yep! Done.